src/Client/StoreOnChainWallet.php

```diff
@@ -0,0 +1,252 @@
+<?php
+
+declare(strict_types=1);
+
+namespace BTCPayServer\Client;
+
+class StoreOnChainWallet extends AbstractClient
+{
+    public function getStoreOnChainWalletOverview(
+        string $storeId,
+        string $cryptoCode
+    ): \BTCPayServer\Result\StoreOnChainWallet {
+        $url = $this->getApiUrl() . 'stores/' .
+                    urlencode($storeId) . '/payment-methods' . '/OnChain' . '/' .
+                    urlencode($cryptoCode) . '/wallet';
+
+        $headers = $this->getRequestHeaders();
+        $method = 'GET';
+
+        $response = $this->getHttpClient()->request($method, $url, $headers);
+
+        if ($response->getStatus() === 200) {
+            return new \BTCPayServer\Result\StoreOnChainWallet(
+                json_decode($response->getBody(), true, 512, JSON_THROW_ON_ERROR)
+            );
+        } else {
+            throw $this->getExceptionByStatusCode($method, $url, $response);
+        }
+    }
+
+    public function getStoreOnChainWalletFeeRate(
+        string $storeId,
+        string $cryptoCode,
+        ?array $metaData
```

Comment:
What's the idea behind this metadata parameter. In the docs I do not see it, there is only a `blockTarget` query parameter which means it would get appended to the url via `?blockTarget=2`.